lib/widget/lottie/web/lottiestate.dart

```diff
@@ -14,115 +18,201 @@ import 'dart:js' as js;
+          ),
+        ),
+      );
     }
-    if (widget.controller.margin != null) {
-      rtn = Padding(padding: widget.controller.margin!, child: rtn);
+
+    if (controller.margin != null) {
+      rtn = Padding(padding: controller.margin!, child: rtn);
     }
+
     return rtn;
   }
 
-  Widget buildLottieHtml(BoxFit? fit) {
-    String source = widget.controller.source.trim();
-    double width = widget.controller.width?.toDouble() ?? 250;
-    double height = widget.controller.height?.toDouble() ?? 250;
-    String _id = widget.controller.id ?? id;
-    String repeat = (widget.controller.repeat == null)
-        ? 'true'
-        : widget.controller.repeat!.toString();
-    String tag = '<div id="$_id"></div>';
-    if (widget.controller.width == null || widget.controller.height == null) {
-      print(
-          'Lottie widget must have explicit width and height for html renderer in the browser (this includes the preview in the editor). You do not need to specify width or height for native apps or for canvaskit renderer. Defaulting to $width for width and $height for height');
+  Widget buildLottieHtml(BoxFit? fit, LottieController controller) {
+    String source = controller.source.trim();
+    double width = controller.width?.toDouble() ?? 250;
+    double height = controller.height?.toDouble() ?? 250;
+    String id = controller.id ?? this.id;
+    String repeat = controller.repeat.toString();
+    String tag = '<div id="$id"></div>';
+
+    if (controller.width == null || controller.height == null) {
+      throw RuntimeError(
+        'Lottie widget must have explicit width and height for html renderer in the browser (this includes the preview in the editor). You do not need to specify width or height for native apps or for canvaskit renderer. Defaulting to $width for width and $height for height',
+      );
     }
 
     if (source.isNotEmpty) {
       // image binding is tricky. When the URL has not been resolved
       // the image will throw exception. We have to use a permanent placeholder
       // until the binding engages
       return JsWidget(
-          id: _id,
-          createHtmlTag: () => tag,
-          data: source,
-          listener: (String msg) {
-            if (widget.controller.onTap != null) {
-              ScreenController().executeAction(
-                  context, widget.controller.onTap!,
-                  event: EnsembleEvent(widget));
-            }
-          },
-          scriptToInstantiate: (String c) {
-            String script =
-                'bodymovin.loadAnimation({container: document.getElementById("$_id"),renderer: "svg",loop: $repeat,autoplay: true,path: "$c"});';
-            if (widget.controller.onTap != null) {
-              script +=
-                  'document.getElementById("$_id").addEventListener("click",() => handleMessage("$_id",""));';
-            }
-            return script;
-          },
-          size: Size(width, height));
+        id: id,
```

Comment:
none of these changes are needed, please cleanup and minimize to only the changes that are required